ext/posix/sys/socket.c

```diff
@@ -103,7 +103,8 @@ pushsockaddrinfo(lua_State *L, int family, struct sockaddr *sa)
 		case AF_UNIX:
 		{
 			struct sockaddr_un *sau = (struct sockaddr_un *)sa;
-			pushlstringfield("path", sau->sun_path, sizeof (sau->sun_path));
+			size_t path_len = strnlen(sau->sun_path, sizeof sau->sun_path);
```

Comment:
`strnlen` is an optional POSIX.2017 extension, and we still support systems back to POSIX.2008.

I would inline a for loop here instead, or you could perhaps add a test for in `lukefile` that sets a macro and adds a full fallback implementation of `strnlen` when the system headers don't provide it so that you get the optimized libc implementation when it is available?

```lua
['posix.sys.socket']    = {
      defines   = {
         HAVE_NET_IF_H           = {checkheader='net/if.h', include='sys/socket.h'},
         HAVE_LINUX_NETLINK_H    = {checkheader='linux/netlink.h', include='sys/socket.h'},
         HAVE_LINUX_IF_PACKET_H  = {checkheader='linux/if_packet.h', include='sys/socket.h'},
         HAVE_STRNLEN            = {checkfunc='strnlen', include='string.h'},
      },
      libraries = {
         {checksymbol='socket', library='socket'},
      },
      sources   = 'ext/posix/sys/socket.c',
   },
```

Reply:
`memchr()` is  POSIX.2008, right?

Reply:
Yep!